### PR TITLE
Add tests for secondary and caching registries

### LINKED ISSUE #2933: Implement DNS/Registry tests
Registry is starting to switch over DNS based communications and gained several capabilities (caching and zone transferring #2703). They are not properly tested ATM. Thanks to #2872, it can and should be tested.

### PR DESCRIPTION
**Pull request by mkykadir**
Fixes #2933 

Depends on #2947 , #2954 and https://github.com/bosagora/localrest/pull/1
after dependencies get merged, this will be rebased (which will reduce the number of commits)

### CODE REVIEW

**Comment by mkykadir** [comment]
Rebased

**Comment by mkykadir** [comment]
Rebased and updated according to the #2961

File: source/agora/test/Base.d [hunk 1]
```diff
@@ -2349,6 +2349,11 @@ public APIManager makeTestNetwork (APIManager : TestAPIManager = TestAPIManager)
       event_handlers : test_conf.event_handlers,
       registry: {
           enabled: true,
+          realm: {
+              authoritative: SetInfo!bool(true, true),
+              primary: SetInfo!string("name.registry", true),
```
Comment (omerfirmak) [inline source/agora/test/Base.d]: Shouldnt this too be an IP?

Comment (mkykadir) [reply source/agora/test/Base.d]: No, this is only used for NS record right now

File: source/agora/test/Base.d [hunk 2]
```diff
@@ -2369,6 +2376,53 @@ public APIManager makeTestNetwork (APIManager : TestAPIManager = TestAPIManager)
     };
     net.createNameRegistry(registry_config, file, line);
 
+    if (test_conf.create_secondary_registry)
+    {
+        Config secondary_registry_config = {
```
Comment (omerfirmak) [inline source/agora/test/Base.d]: `makeRegistryConfig` ?

**Review by omerfirmak — APPROVED** [review]

**Pull request merged**